### ISSUE TITLE
Group Props: container-type messes with layouts

### ISSUE THREAD

**Issue by kzlar** [tamagui/tamagui#1823]
### Current Behavior

See thread on discord: https://discord.com/channels/909986013848412191/974145843919716412/1169724541354905660
Basically when using group props on web, the following css rule gets set `container-type: inline-size` which causes some layout issues for buttons. Native seems fine.

https://github.com/kzlar/tamagui/commit/f560f531b52c743ef176335cd68151e5ee1524f5
In the commit above I have a repro and a workaround I hacked together, if you think this is something you want to take a solution I can clean it up and submit a PR.

### Tamagui Version

```markdown
latest
```

### Platform (Web, iOS, Android)

```markdown
web
```

### Reproduction

```markdown
https://github.com/kzlar/tamagui/commit/f560f531b52c743ef176335cd68151e5ee1524f5

To switch between inline-size and normal comment the line with `webContainerType` in the kitchen sink app tamagui config.
```

**Comment by natew** [comment]
The code looks good and I can probably merge a PR like that, but the discord is hard to follow, can you sum up what is breaking? I'm reading up on `container-type`.

**Comment by kzlar** [comment]
Sure. The gist is that adding a group prop to buttons is breaking layout. For example take the following button:
```
<View style={{ width: '100%', height: '100%', padding: 50, flexDirection:'row', backgroundColor:'black' }}>
      <Button group='ButtonContextGroup'  bc='green' hoverStyle={{bc: 'purple'}} forceStyle='hover'>
        <Button.Text 
        $group-ButtonContextGroup-hover={{
          color: 'red',
        }} 
        fontFamily={'$body'} 
        >
          ??
      </Button.Text>
      </Button>
      <Button group='ButtonContextGroup'  bc='green'>
        <Button.Text 
        $group-ButtonContextGroup-hover={{
          color: 'red',
        }} 
        fontFamily={'$body'} 
        >
          ??
      </Button.Text>
      </Button>
      <Button group='ButtonContextGroup'  bc='green'>
        <Button.Text 
        $group-ButtonContextGroup-hover={{
          color: 'red',
        }} 
        fontFamily={'$body'} 
        >
          ??
      </Button.Text>
      </Button>
    </View>
```
And then switch between defining and not defining the following (incl. my change): `webContainerType: 'normal'` 
You'll see the correct layout (with defining the above):
<img width="242" alt="image" src="https://github.com/tamagui/tamagui/assets/120426485/d098449f-0c50-4890-aa3f-22b66fe994d7">
And wrong layout (without defining the above):
<img width="195" alt="image" src="https://github.com/tamagui/tamagui/assets/120426485/a3392cc5-c221-4342-beea-979c4397c3a9">

While not directly related to this issue, you can also see the other thing we discussed on discord where the `forceStyle='hover'` is not honored on the text component.

**Comment by natew** [comment]
Sounds good, I've implemented your change in cc7654a7f13badfbf7270d184322ae44ad387d2e and am going to switch the default behavior to `normal`, since groups are in beta we can make breaking changes in minor versions and I'll just call it out.

**Comment by kzlar** [comment]
Cool, thanks Nate! 🎉

**Comment by bryanmylee** [comment]
For anyone coming to this in the future and not really understanding the need for `inline-size`, it's required if we want to do container size queries i.e. style children based on the size of a parent container. Therefore, it's not advisable to switch to `config.settings.webContainerType: "normal"`.

Instead, if you need a specific component to be `container-type: normal`, set:

```ts
"$platform-web": {
  containerType: "normal",
},
```

**Comment by natew** [comment]
You can also set `containerType` directly on the regular props btw.
